projectile.el

```diff
@@ -657,7 +657,7 @@ The cache is created both in memory and on the hard drive."
   "Add the currently visited file to the cache."
   (interactive)
   (let ((current-project (projectile-project-root)))
-    (when (gethash (projectile-project-root) projectile-projects-cache)
+    (when (and (gethash (projectile-project-root) projectile-projects-cache) (buffer-file-name (current-buffer)))
```

Comment:
I'd move the file check before the cache check. Also `(buffer-file-name (current-buffer))` can be replaced with simply `buffer-file-name`. No need for two function calls. I see the same was used a line below, which can be optimized.